### ISSUE TITLE
v2.16.1 - Passing empty array to chunks option does not create html file

### ISSUE THREAD

**Issue by eladlevy** [ampedandwired/html-webpack-plugin#315]
In version 2.16.0 the following webpack configuration created the html file as expected:
```
new HtmlWebpackPlugin({
            filename: '/path/to/file/generated_page.html',
            template: 'some_page.html',
            chunks: [],
            inject: false
        }),
```

But in 2.16.1 the html file was not generated. Only omitting chunks option resolved the issue:
```
new HtmlWebpackPlugin({
            filename: '/path/to/file/generated_page.html',
            template: 'some_page.html',
            inject: false
        }),
```

I would expect that passing empty array will be equivalent to not passing chunks option like in older versions. Is that a right assumption?

**Comment by graingert** [comment]
interestingly this works in dev mode.

Also passing an array with a non existent module in does not work.

**Comment by jantimon** [comment]
Oh this is probably because of the hot module replacement "fix":

https://github.com/ampedandwired/html-webpack-plugin/compare/v2.16.0...v2.16.1#diff-168726dbe96b3ce427e7fedce31bb0bcR76

I guess `isHotUpdateCompilation(assets)` returns true if the chunks array is empty.
I am on the road right now but will look into it in the next days

**Comment by graingert** [comment]
@jantimon https://github.com/ampedandwired/html-webpack-plugin/pull/319

**Comment by jantimon** [comment]
Fixed in v2.16.2
